### ISSUE TITLE
Docker version is not updated

### ISSUE THREAD

**Issue by canne** [SignalK/signalk-server-node#900]
Hi, I have installed signal-server on Docker (on Win10), the latest is from 4 months, https://hub.docker.com/layers/signalk/signalk-server/latest/images/sha256-1ee27346cdc1854c895c06db2db28486d59b8a65e0019cbcd8fe94e4aa75747b
At startup I get:
<img width="516" alt="Annotation 2019-11-23 220518" src="https://user-images.githubusercontent.com/6381128/69485376-ecb0ea80-0e3e-11ea-8bdb-d0050bc4956d.png">

Is there a plan to update to 1.18.0 and is there a way in this case to update the image (like reinstalling a new image) without losing data and configuration (like boat settings and home directory data, perhaps with an external volume)?

**Comment by tkurki** [comment]
Version 1.18.0 is available on Docker Hub: https://hub.docker.com/r/signalk/signalk-server/tags but for some reason the `latest` tag is not updated, I need to look into that.

But I have not bothered to work on the automatic update process for running under Docker because very few people use it. 

The idea is that you would run the server so that the `.signalk` directory (and if you log data the directory that you use for that) would be mounted from the host filesystem. This way you can stop the server, pull a new image from Docker Hub and restart it with all your data on the host system.

If you have not done so I suggest you use `docker cp` to copy your settings from inside the container to the host system and then update. 

https://docs.docker.com/engine/reference/commandline/cp/

In versions <1.17.0 you can find the settings in /home/signalk, in newer versions they are under /home/node.

[Here](https://github.com/tkurki/marinepi-provisioning/blob/master/roles/signalk-docker/templates/docker-compose.j2#L1-L24) is a docker-compose that mounts the necessary directories and exposes ports for running sk server. You can do the same thing from the command line with appropriate options, Docker on Win has no doubt its own way for doing this. Note that is is not yet updated for >= v 1.17.

**Comment by canne** [comment]
Thanks, I'll try that out on RPI. The screenshot was from Docker on Windows. Can you please update / keep it updated as well.

**Comment by tkurki** [comment]
Travis automatically builds linux-armhf and linux-amd64 images. 

And it is supposed to build a multiarch image as well, so that you can just pull a specific version and it will automatically pull the correct arch version, but last time I looked it did not work correctly.

**Comment by canne** [comment]
This happens to me with Travis more and more often, timeouts. Free users are at the end of the line, I guess. Need to go to Travis GUI and relaunch the build. Not very automatic. `armhf` in Travis? Thanks, I need to take a look again, I somehow have missed it.

**Comment by tkurki** [comment]
https://github.com/SignalK/signalk-server-node/blob/master/.travis.yml#L21-L28

**Comment by canne** [comment]
Thanks for the link; will try the multiarch out on Travis. 
I have successfully implemented the method you described regarding the home directory for the version 1.1.17 I have got. There is not many differences to Windows installation compared to RPI, I've documented they way I did the home directory mount in Windows Docker Desktop, [here](https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:supplementary_software:signalk:a3#signal_k_node_server_for_nodejs_as_docker-image) (server has some intermittent issues when this is written, sorry).
Remains the version 1.1.18 update for Windows Docker and the ticket can be resolved without further ado, many thanks for your help!
